plugins/XmlMaterialProfile/XmlMaterialUpgrader.py

```diff
@@ -5,10 +5,19 @@
 
 from UM.VersionUpgrade import VersionUpgrade
 
-from cura.CuraApplication import CuraApplication
 from .XmlMaterialProfile import XmlMaterialProfile
 
 
+CARTESIO_VARIANT_ID_TO_NEW = {
+    "0.25 mm": {"id": "0.25mm thermoplastic extruder",
+                "alternative_ids": "0.25 mm"},
+    "0.4 mm": {"id": "0.4mm thermoplastic extruder",
+               "alternative_ids": "0.4 mm"},
+    "0.8 mm": {"id": "0.8mm thermoplastic extruder",
+               "alternative_ids": "0.8 mm"},
+}
```

Comment:
Also related to below: This feels super messy to me to add all this 3rd party conversion stuff into our code base. Is this not something we can wait until 4.0 to do (or rather, not do, and be ok with breaking changes)?

Reply:
Wasn't the plan to put the actual alternative_ids in the material files?

Reply:
@ChrisTerBeke The plan was to add `alternative_ids`. This is XML material file upgrade, which also needs to upgrade the old files to have `alternative_ids`. Since we are already doing this, it's better to update the id as well.

@ianpaschal I discussed with @diegopradogesto and this can break for, for example, UM3 with a custom PVA because generic_pva has those missing variants of cartesio, so we think it would be nice to have it fixed, although I totally agree with you that this is super messy not to mention that we are using the XML schema version as the `settings_version` of the XML material.

Reply:
If you guys think it's better not to have this "fix", I'm ok with it.

Reply:
Mm this is getting out of hand :P In that case I'd say skip this fix and just say that IDs are supposed to be immutable.

Reply:
Yeah I don't think IDs should be mutable. I also think the human-readable thing we do everywhere is really bad practice because we end up with all sorts of IDs that don't reflect the data anymore but are just super confusing.

These should have immutable UUID v4 or whatever.

Reply:
I agree that ID should be immutable otherwise let name it different. The CARTESIO_VARIANT_ID_TO_NEW also should be removed, If we want to make fix then I see only create the profile which does not have any specific information. Otherwise what if CARTESIO will not make a fix then these code lines will be forever in Cura.

Reply:
I'm also OK with not fixing it. The only problem is that users will see the configuration errors even if they don't have Cartesio printers, just because this `machine` section is in a lot of material profiles, so for instance `generic_pva` (just to mention one).
We did a mistake approving the original PR that caused this, but now we need to agree in a solution, either live with it or fix it in a horrible way.